packages/xrpl/src/Wallet/fundWallet.ts

```diff
@@ -83,205 +91,134 @@ const MAX_ATTEMPTS = 20
  * and that wallet's balance in XRP.
  * @throws When either Client isn't connected or unable to fund wallet address.
  */
-// eslint-disable-next-line max-lines-per-function -- All lines necessary
 async function fundWallet(
   this: Client,
   wallet?: Wallet | null,
-  options?: {
-    faucetHost?: string
-    faucetPath?: string
-    amount?: string
-  },
+  options: FundingOptions = {},
 ): Promise<{
   wallet: Wallet
   balance: number
 }> {
   if (!this.isConnected()) {
     throw new RippledError('Client not connected, cannot call faucet')
   }
+  const createWallet = wallet === null
 
   // Generate a new Wallet if no existing Wallet is provided or its address is invalid to fund
-  const walletToFund =
+  const walletToFund: Wallet =
     wallet && isValidClassicAddress(wallet.classicAddress)
       ? wallet
       : Wallet.generate()
 
   // Create the POST request body
-  const postBody = Buffer.from(
-    new TextEncoder().encode(
-      JSON.stringify({
-        destination: walletToFund.classicAddress,
-        xrpAmount: options?.amount,
-      }),
-    ),
-  )
+  const postBody: FaucetRequestBody = {
+    destination: walletToFund.classicAddress,
+    xrpAmount: options.amount,
+  }
 
   let startingBalance = 0
-  try {
-    startingBalance = Number(
-      await this.getXrpBalance(walletToFund.classicAddress),
-    )
-  } catch {
-    /* startingBalance remains '0' */
+  if (!createWallet) {
+    try {
+      startingBalance = Number(
+        await this.getXrpBalance(walletToFund.classicAddress),
+      )
+    } catch {
+      /* startingBalance remains '0' */
```

Comment:
is there supposed to be code here for catch?

Reply:
No there was before.

Reply:
Relooking at this i bet it always fails because there is no `this`

Reply:
you meant if there wasn't a catch logic before?

Reply:
There was a catch before

Reply:
nit: Separate issue on this line - it could fail due to network connectivity instead of ledger reasons - in which case a previously funded account would have the same funds as before which can be greater than zero.
```suggestion
      /* startingBalance remains what it was previously */
```